api/app/models/concerns/notification_preferences.rb

```diff
@@ -0,0 +1,66 @@
+module Concerns
+  module NotificationPreferences
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :notification_preferences, dependent: :destroy, autosave: true
+
+      scope :with_digest_for_frequency, lambda { |frequency|
```

Comment:
What these scopes are intended for is not very clear to me. Can the naming be improved here at all?